biomage/utils/data.py

```diff
@@ -25,6 +97,51 @@ def modified_records(item, target_table, config):
                         }
                     },
                 }
+            },
+            # Modify project id to point to sandboxed project
+            "projectId": {
+                'S': f"{extra['sandbox_id']}-{item['projectId']['S']}",
+            },
+            # Add user id to experiment
+            "rbac_can_write": {
+                "SS" : [
+                    *item["rbac_can_write"]["SS"],
+                    add_env_user_to_experiment(extra['username'], config)
```

Comment:
this function name is completely misleading it adds nothing, just returns a username. As mentioned this can be done recursively in a short function (which already existed `add_user_to_rbac`) avoiding the need of this dictionary recreations which just clutter the method.

Reply:
Changed accordingly